data/cookbook/networking/tcp-server-lwt/00-lwt.md

```diff
@@ -0,0 +1,77 @@
+---
+packages:
+- name: lwt
+  version: "5.7.0"
+- name: logs
+  version: "0.7.0"
+sections:
+- filename: main.ml
+  language: ocaml
+  code_blocks:
+  - explanation: |
+      Defines some constants. The `listen_address` is typically `Unix.inet_addr_loopback`, `Unix.inet_addr_any`. Other values may be used to listen only on one network interface. The `(let*)` operator permits the chaining of multiple Lwt statements.
+    code: |
+      let (let*) = Lwt.bind
+
+      let listen_address = Unix.inet_addr_loopback
+      let port = 9000
+      let backlog = 10
+  - explanation: |
+      This `loop` function loop forever a given Lwt promise.
+    code: |
+      let rec loop f =
+        let* () = f () in
+        loop f
+  - explanation: |
+      This defines a function which will handle the connection with a single client. `ic` and `oc` are input and output channels that can be used with `Lwt_io` functions.
+    code: |
+      let rec handle_connection ic oc =
+        let* () = Lwt_io.write_line oc "Give me your name:" in
+        let* line = Lwt_io.read_line_opt ic in
+        match line with
+        | Some line' ->
+           let* () = Lwt_io.write_line oc ("Hello, " ^ line') in
+           handle_connection ic oc
+        | None ->
+           Logs_lwt.info (fun m -> m "Connection closed")
+  - explanation: |
+      This defines a function which "accept" a new connection, and run `handle_connection` on this connection. `Lwt.on_failure` returns immediately and execute this function in paralel with the other tasks.
+    code: |
+      let accept_connection socket_fd =
+        let* conn = Lwt_unix.accept socket_fd in
+        let fd, _client_addr = conn in
+        let ic = Lwt_io.of_fd ~mode:Lwt_io.Input fd in
+        let oc = Lwt_io.of_fd ~mode:Lwt_io.Output fd in
+        Lwt.on_failure
+          (handle_connection ic oc)
+          (fun exc -> Logs.err
+                        (fun m -> m "%s" (Printexc.to_string exc) ));
+        Logs_lwt.info (fun m -> m "New connection")
+  - explanation: |
+      The main function initialise the socket that will serve for the server and loop forever the `accept_connection` function.
```

Comment:
```suggestion
      The main function initialises the socket that will serve for the server and loop forever the `accept_connection` function.
```
Is there another verb we can use here other than "serve" as well? So it's not "serve for the server"? Also, should it also be "loop forever through the..."